### PR TITLE
Fix relocation for esmf.mk

### LINKED ISSUE #92: `esmf.mk` is not completely rewritten?
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

The `esmf.mk` file that is installed in `$PREFIX/lib` contains a number of unreplaced build environment paths. I am not sure if all of them *should* be replaced, but we might need to add some hints to the `meta.yaml` for that to happen.

I also wonder why the rewriting doesn't happen. Isn't conda-build supposed to find text files and replace the paths automatically?

### Installed packages

```shell
-
```

### Environment info

```shell
-
```

### PR DESCRIPTION
**Pull request by zklaus**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [x] ~Reset the build number to `0` (if the version changed)~
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Fixes #92 

This adds rewriting of `$BUILD_PREFIX` to `$PREFIX` in `lib/esmf.mk`. This should help with downstream users of `esmf.mk`, such as `esmpy`.

### CODE REVIEW

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
@conda-forge-admin, please rerender

**Comment by zklaus** [comment]
I think this could work. What do you think, @conda-forge/esmf?

**Review by xylar — APPROVED** [review]
Let's see how it goes! If this works, we should try it in netcdf-fortran, too.

**Comment by xylar** [comment]
@zklaus, the ppc64le jobs are just timing out.  Could you delete a few more tests here to make them work?

**Comment by zklaus** [comment]
> @zklaus, the ppc64le jobs are just timing out. Could you delete a few more tests here to make them work?

Actually, I am afraid that's at least not the full story. I got suspicious because we skip all testing for mpich on ppc64le anyways. A closer look reveals that now it doesn't even mange to finish `make esmflib`, c.f. [nompi](https://app.travis-ci.com/github/conda-forge/esmf-feedstock/jobs/591235055#L3868) and [mpich](https://app.travis-ci.com/github/conda-forge/esmf-feedstock/jobs/591235054#L4060) (apparently in this attempt, the openmpi part didn't run at all due to travis flakiness). I don't think that is related to the change I made, because that is a change after the make, so maybe something from the rerender slows things down overall?

**Comment by xylar** [comment]
Ah, the new gnu could indeed explain that.  How frustrating.

Could we switch to cross-compiling and let the tests go?

**Comment by zklaus** [comment]
Yeah, I am in favor of that. Do you know how to do that (I don't atm)? Feel free to push to this branch.

**Comment by xylar** [comment]
I believe I know how to do that but don't have time just now.  I'll push some changes soon...

**Comment by xylar** [comment]
Let's try just building on Azure first.  If it's too slow, we can switch to cross-compiling as well.  See https://github.com/conda-forge/gdal-feedstock/pull/566 for some discussion.

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

**Comment by xylar** [comment]
Okay, 5 hours in, I'm pulling the plug on azure without cross-compiling.  Way too slow.

**Comment by xylar** [comment]
@conda-forge-admin, please rerender

File: recipe/build.sh [hunk 1]
```diff
@@ -31,6 +31,13 @@ fi
 
 echo ESMF_F90COMPILEOPTS=${ESMF_F90COMPILEOPTS}
 
+if [[ "${CONDA_BUILD_CROSS_COMPILATION:-}" == "1"  && "${target_platform}" == "linux-aarch64" ]]; then
+  # having trouble with -m64 on aarch64 when cross-compling and this is the "fix"
+  # that happens when not cross-compiling
+  export ESMF_ABI=32
+fi
```
Comment (xylar) [inline recipe/build.sh:38]: I've got to say I'm not very happy with this fix (let's see if it even works) but this seems to be why we weren't seeing this problem when we weren't cross-compiling.  ESMF seems to think aarch64 is 32 bit. See a recent successful Travis build:
https://app.travis-ci.com/github/conda-forge/esmf-feedstock/jobs/590175359

**Comment by xylar** [comment]
`ESMF_ABI` didn't stick.

**Comment by xylar** [comment]
Not having much luck getting ESMF to cross-compile so far.

**Comment by xylar** [comment]
Okay @zklaus, I think we're good.  Let me know what you think and feel free to merge once CI suceeds.

**Comment by zklaus** [comment]
Looking good. Thanks for all the work, @xylar!

**Comment by xylar** [comment]
Great, I think this is a reasonable solution.

**Pull request merged**